streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -220,7 +220,7 @@ boolean allPartitionsBuffered() {
 
     void close() {
         clear();
-        partitionQueues.clear();
+
         streamTime = RecordQueue.UNKNOWN;
```

Comment:
@guozhangwang we should also reset the `partitionTime` for the `RecordQueue`s in `#clear`?

Reply:
I think that's legitimate

Reply:
`clear` is also called in `suspend` in which we should not clear the partitionTime, but I think we should reset the `partitionTime` in close.

Reply:
Sorry for the late comment as the PR is already merbe, but if we don't clean the `partitionQueues`, why would we need to reset `streamTime` or `partitionTime` ? I we keep all records in the buffer all know times stay valid, don't they?

Reply:
Similar to resetting `partitionTime`, we should also reset `streamTime` in `PartitionsGroup#clear()` below though?

Reply:
Opened a minor PR: https://github.com/apache/kafka/pull/8250 (let me know what you think)